### ISSUE TITLE
MxToolbox DKIM Signature Body Hash Did Not Verify

### ISSUE THREAD

**Issue by qbitroot** [Mailu/Mailu#2330]
Hello! I installed **Mailu 1.9** with latest software on Debian 11 VPS server using **Docker 20.10.14** and **Docker Compose v2.4.1** with **Roundcube** webmail.

## Description
When I send email to `ping@tools.mxtoolbox.com` to test if the mail server config is valid, it reports that "**Body Hash Did Not Verify**". (I have added DNS entries for DKIM key). DKIM keys alone (just DNS) are valid though.
In this email the Body Hash is `vPHMrXurm0Buq8kkhXMITOdhG6u0qp7fj0OAg/j8d+M=`.
Full report: https://mxtoolbox.com/deliverability/8dd8360f-2808-4ad4-a780-5cb3c09a8c32
I looked for similar issues but not found a fix.

## Replication Steps

1. Install Docker, Docker-Compose, Mailu.
2. Create DKIM key via admin panel.
3. Setup DNS records - MX, SPF, DMARC, PTR and **DKIM**.
4. Check if you can send  at least to SPAM folder and receive emails.
5. Send any email to `ping@tools.mxtoolbox.com`.
6. Get a report in the inbox. You should see "**❌ DKIM Not Authenticated**" error because of the "invalid body hash".

## Expected behaviour
According to mxtoolbox.com, the "Expected Body Hash" for the message below is `47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=`. In past I configured the same server using poste.io freeware, which Mailu aims to substitute and it showed no errors.

## Logs
**Report**: https://mxtoolbox.com/deliverability/8dd8360f-2808-4ad4-a780-5cb3c09a8c32
There are no errors in container logs.
Headers got from MxToolbox:
```
From admin@virbox.net  Wed Apr 27 17:50:21 2022
Return-Path: <admin@virbox.net>
X-Original-To: ping@tools.mxtoolbox.com
Delivered-To: tools@tools.mxtoolbox.com
Received: from mail.virbox.net (virbox.net [157.230.28.179])
	(using TLSv1.2 with cipher ADH-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by tools.mxtoolbox.com (Postfix) with ESMTPS id 577EDB5F20
	for <ping@tools.mxtoolbox.com>; Wed, 27 Apr 2022 17:50:21 +0000 (UTC)
MIME-Version: 1.0
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=virbox.net; s=dkim;
	t=1651081820;
	h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
	 to:to:cc:mime-version:mime-version:content-type:content-type;
	bh=vPHMrXurm0Buq8kkhXMITOdhG6u0qp7fj0OAg/j8d+M=;
	b=A+FIHxEGC45I/6uufCQ5y8WFIMs2x/nB/a0QBmTSYQ2O/w9K+CGTtqo+KpqoxUfGEt+H+4
	IkSSWnh02FIJARlXs3p6JvlWWLwZxu5DOcwPmZ8lVY+Dv46X5uPqZBu20tGYmkNLy5dOo8
	/ctMgDsldfKZvDfLW3EewUknx/kqzlz1e51rOejzwyqm1H7xywiHCE5zo/SwWgLXFAUnOs
	2UXwj5LH2ppgV0qVYd2e2BeoSdlASnSCXuhzej1fc/VtywLzjYZjY9Gw7K1XQafAj7ToO+
	BB2YBwPL2hvKopHpPuKe7t7bL9xvVfdjnFyE2/eMIKEBzQtTi2fwlB/pW2vXaA==
Date: Wed, 27 Apr 2022 17:50:20 +0000
From: Qbit <admin@virbox.net>
To: Ping <ping@tools.mxtoolbox.com>
Subject: Testing the server
Message-ID: <12967bfffbed808400cf3b134672a29d@virbox.net>
X-Sender: admin@virbox.net
Content-Type: multipart/alternative;
 boundary="=_cf09ec6238d0852de338ec0494815fcf"
X-Spamd-Bar: /
Authentication-Results: mail.virbox.net;
	auth=pass smtp.mailfrom=admin@virbox.net
```
The email I sent (.eml from Roundcube):
```
MIME-Version: 1.0
Date: Wed, 27 Apr 2022 17:50:20 +0000
From: Qbit <admin@virbox.net>
To: Ping <ping@tools.mxtoolbox.com>
Subject: Testing the server
Message-ID: <12967bfffbed808400cf3b134672a29d@virbox.net>
X-Sender: admin@virbox.net
Content-Type: multipart/alternative;
 boundary="=_cf09ec6238d0852de338ec0494815fcf"

--=_cf09ec6238d0852de338ec0494815fcf
Content-Transfer-Encoding: 7bit
Content-Type: text/plain; charset=US-ASCII;
 format=flowed



I hope that DKIM hash will be valid
--=_cf09ec6238d0852de338ec0494815fcf
Content-Transfer-Encoding: quoted-printable
Content-Type: text/html; charset=UTF-8

<html><head><meta http-equiv=3D"Content-Type" content=3D"text/html; charset=
=3DUTF-8" /></head><body style=3D'font-size: 12pt; font-family: Arial,Helve=
tica,sans-serif'>
<p>I hope that DKIM hash will be valid</p>
</body></html>

--=_cf09ec6238d0852de338ec0494815fcf--
```
I would accept that it is an error on the mxtoolbox side, but all emails go to SPAM with Gmail. poste.io worked fine however. I will provide additional data if needed.

**Comment by nextgens** [comment]
I can reproduce on my own setup. I think it may be a bug on the mxtoolbox side.

There are things we could do, starting with avoiding the duplication of the fields... our current DKIM config looks sub-optimal but I don't think it's broken.

Here when I send an email to gmail it does tell me that the DKIM verification was successful.

**Comment by qbitroot** [comment]
Thanks for the reply!

> Here when I send an email to gmail it does tell me that the DKIM verification was successful.

Didn't know about the ability to view checks from Google side, it is a good idea. Indeed, even when email arrived to SPAM, clicking "Show original" in Gmail shows **DKIM: 'PASS' with domain virbox.net**.

Most likely emails go to spam now because my mail server was up for a week and then **was down for like 10 days** and up again now. It could corrupt its trust score and therefore spam problem is not related.

But the issue with MxToolbox is still open.

**Comment by qbitroot** [comment]
As a sanity check, I configured poste.io again and edited DKIM. It sent to spam and MxToolBox showed DKIM fail too. So it doesn't matter whether I use poste or Mailu.

Moreover, I tried to send to `ping@tools.mxtoolbox.com` from Gmail, and it, surprisingly, showed the DKIM hash error too and **the same "Expected Body Hash"** - `47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=` ! So it should be the bug on MxToolbox side. So this software is great!
